torchci/components/WorkflowDispatcher.tsx

```diff
@@ -141,3 +140,72 @@ export default function WorkflowDispatcher({
     </div>
   );
 }
+
+export function SingleWorkflowDispatcher({
+  sha,
+  jobName,
+}: {
+  sha: string;
+  jobName: string;
+}) {
+  let [runMoreJobsClicked, setRunMoreJobsClicked] = useState(false);
+
+  let session: any = useSession();
+  const router = useRouter();
+
+  let { repoOwner, repoName } = router.query;
+
+  if (!repoOwner) {
+    repoOwner = "pytorch";
+    repoName = "pytorch";
+  }
+
+  // extract workflow key from the jobName
+  const workflow = jobName.split(new RegExp("[/-]"))[0]?.toLowerCase()?.trim();
+
+  const repo = `${repoOwner}/${repoName}`;
+  if (
+    !session ||
+    !session.data ||
+    !session.data["accessToken"] ||
+    !(repo in SUPPORTED_WORKFLOWS) ||
+    !workflow ||
+    !(workflow in SUPPORTED_WORKFLOWS[repo])
+  ) {
+    return <></>;
+  }
+
+  const accessToken = session.data["accessToken"];
+
+  // avoid commit data fetching if the user hasn't clicked the button
+  if (!runMoreJobsClicked) {
+    return (
+      <div>
+        <button onClick={() => setRunMoreJobsClicked(true)}>
+          Run more jobs?
+        </button>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <WithCommitData
+        sha={sha}
+        repoOwner={repoOwner as string}
+        repoName={repoName as string}
```

Comment:
If I correctly understood your question, the only way to call react hook conditionally, is to separate it via component boundary (and render the whole component conditionally).
```
  const { data, error } = useSWR(
```

It's just the limitation on how react hooks work, the component expects the same hooks in to be called, in the same order.

I could probably use `fetcherWithToken` instead of the `useSWR` hook, but the component would lose some interactivity (like the autorefresh).